conda/cli/main_env_create.py

```diff
@@ -141,70 +154,97 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
     prefix_data.validate_path()
     prefix_data.validate_name()
 
-    # TODO, add capability
-    # common.ensure_override_channels_requires_channel(args)
-    # channel_urls = args.channel or ()
-
-    result = {"conda": None, "pip": None}
+    # Convert all environment configuration to environment variables for fresh_context
+    # NOTE: Only channels and environment variables are currently supported in
+    # environment.yaml files.
+    env_vars = {}
+    # Add environment variables from YAML (convert all values to strings)
+    if env.variables:
```

Comment:
These `env.variables` are not the same as env vars for the purpose of configuration. They are just environment variables that the user wants setup in their environment.

Reply:
ok so, after chatting with @ryanskeith, I think we do want to include these here. For example, its pretty reasonable for someone to have an environment.yml

```
name: myenv
channels:
  - conda-forge
dependencies:
  - python
variables:
  VARIABLE: value
  CONDA_SOLVER: libmamba
prefix: /something/miniconda/envs/test
```

On the main branch of conda this environment.yml environment configuration variables (`CONDA_SOLVER`) will not get applied on initial environment creation. But, this configuration will get persisted to the created environment.

So, this line has a bit of change in behaviour. Now, the environment variables defined here will get included into the context on environment creation.